packages/playwright-core/src/server/console.ts

```diff
@@ -17,25 +17,25 @@
 import { SdkObject } from './instrumentation';
 import type * as js from './javascript';
 import type { ConsoleMessageLocation } from './types';
-import type { Page } from './page';
+import { Page } from './page';
 
 export class ConsoleMessage extends SdkObject {
   private _type: string;
   private _text?: string;
   private _args: js.JSHandle[];
   private _location: ConsoleMessageLocation;
-  private _page: Page;
+  private _page: Page | undefined;
 
-  constructor(page: Page, type: string, text: string | undefined, args: js.JSHandle[], location?: ConsoleMessageLocation) {
-    super(page, 'console-message');
-    this._page = page;
+  constructor(parent: SdkObject, type: string, text: string | undefined, args: js.JSHandle[], location?: ConsoleMessageLocation) {
```

Comment:
Since we do not have ConsoleMessageDispatcher anymore, we might as well drop `SdkObject` inheritance in this class. This should make your change easier.